Pause Menu

AmberSolstice opened the pull request:
### Pause Menu Implementation

This pull request introduces a functional Pause Menu system, allowing players to pause game play and access options such as resume, settings, or returning to the main menu. 

#### Features Implemented

- Pause functionality via Esc key or UI button
- Tab-based menu system with Pause tab
- Resume, Quit to Main Menu, and Exit Game options
- Game play frozen using get_tree()paused = true
- Hot bar hidden while menu is open
- Player input disabled while paused
- Menu remains active using pause_mode = Process

#### Notes

- Pause state is reset before scene transitions
- Buttons set to process while paused
- Player script respects paused state
- Hot bar visibility toggled on pause
- Input handled


Closes #22

Amber-Elara-Solstice committed:
feat: Changed how the inventory is showen adding a tab altho it is not working yet and i really should of used a tutorial or did my homework before trying this

AmberSolstice committed:
fix: Scaling has been ajusted and the tabs are now visable it was an issue with child nodes being different from parents in a messed up way

AmberSolstice committed:
feat: Pause menu tab and outline done

AmberSolstice committed:
feat: Added going back to menu and cloising thge game in game so you can navigate around the game with hsortcuts also

hartgit reviewed (changes requested):
I found a bug, when the inventory is open and the Esc key is pressed to close it, the cursor disappears.

It works well otherwise, all the buttons seem functional. Great job!

kblack89 reviewed (changes requested):
Most of this works well, just a few issues such as being able to water and plants still growing on pause menu and escape not being set up. The function of the menu popping up and working with the main menu is good.

verstraate reviewed (changes requested):
## Review for #46 
I like the design and how it matches the other menus that have been implemented. It's very helpful that there is a tab system at the top of both the inventory and pause menu, which is a great UX consideration!

### Issues
Like @kblack89 mentioned, there are times when you can still do stuff whilst in-game in the pause menu. I couldn't get the player to plant seeds while in the menu, but I was able to water the dirt. Also, the player can still pick up items in the menu, so long as they're within range of the item.

I also noticed the same issue as @hartgit mentioned, where the cursor disappears on closing the pause menu. The only thing different that I noticed is that this happens whenever you close either the inventory or pause menu, no matter what you did in those menus. However, I found that when you go to the main menu from the pause menu this fixes the issue. Most likely the menus somehow change something within the main game scene and changing scenes fixes it.

### Suggestions (not required)
There are some features that would be nice to have, but potentially in a future sprint. First, being able to press `ESC` to enter and exit the pause menu (and any other UI menu; not relevant here) would be a great UX addition. Player's are familiar with the concept of `ESC` being the default "exit" method for both exiting the main game and its menus. Also, whilst I am fully aware that they aren't currently implemented, having a settings option in the pause menu would be nice.

### Final comment
Other than that, I like what you've done here so far and the only problems I have should be reasonably straight forward, given that you've already addressed some of these issues elsewhere!

AmberSolstice committed:
fix: curser issue resolved, escape notworking resolved, things not pausing resolved

AmberSolstice commented:
### Review Request Update

I've pushed a fix addressing all the issues raised in the previous review's:

- **Cursor disappearing** – Now properly resets visibility when closing menus  
- **Escape key handling** – ESC now correctly toggles the pause menu and works with the inventory  
- **Game play actions while paused** – Interactions like watering, item pickups, and growth are now blocked while paused  
- **Pause state propagation** – Ensured `pause_mode = Process` is correctly set on UI elements and necessary scripts

verstraate reviewed (approved):
## Rereview for #46 
Everything has been changed as per requested! However, I was wondering if the act of pressing `ESC` whilst in the inventory and it going to the pause menu was intentional. Either way I think it's fine and this is good to merge.

hartgit reviewed (approved):
Bug fixes have been made, looks good.
